typescript/cli/src/context.ts

```diff
@@ -116,12 +116,71 @@ export async function getContext<P extends ContextSettings>({
   } as CommandContext<P>;
 }
 
+/**
+ * Retrieves dry-run context for the user-selected command
+ * @returns dry-run context for the current command
+ */
+export async function getDryRunContext<P extends ContextSettings>({
+  chainConfigPath,
+  chains,
+  coreConfig,
+  keyConfig,
+  skipConfirmation,
+}: P): Promise<CommandContext<P>> {
+  const customChains = readChainConfigsIfExists(chainConfigPath);
+
+  let coreArtifacts = undefined;
+  if (coreConfig) {
+    coreArtifacts =
+      (await runDeploymentArtifactStep({
+        artifactsPath: coreConfig.coreArtifactsPath,
+        message:
+          coreConfig.promptMessage ||
+          'Do you want to use some core deployment address artifacts? This is required for PI chains (non-core chains).',
+        skipConfirmation,
+      })) || {};
+  }
+
+  const multiProvider = getMultiProvider(customChains);
+
+  if (!chains?.length) {
```

Comment:
What happens if multiple chains are provided via the param here?

Reply:
The first provided chain will be forked if there are multiple.

This feature's next iteration will include multi-chain support and avoid naively choosing the first selection